Classes/Private/Libxml2/Data/CommentNode.swift

```diff
@@ -0,0 +1,93 @@
+import Foundation
+
+extension Libxml2 {
+    /// Comment nodes use to hold HTML comments like this: <!-- This is a comment -->
+    ///
+    class CommentNode: Node {
+
+        var text: String
+
+        init(text: String) {
+            self.text = text
+
+            super.init(name: "text")
+        }
+
+        override func customMirror() -> Mirror {
+            return Mirror(self, children: ["type": "text", "name": name, "text": text, "parent": parent.debugDescription], ancestorRepresentation: .Suppressed)
+        }
+
+        /// Node length.
+        ///
+        override func length() -> Int {
+            return text.characters.count
```

Comment:
Alternatively, we could leave the length as it is, but ignore it for the editing logic, since editing comments could come in handy.